homeassistant/components/xiaomi_miio/fan.py

```diff
@@ -1103,6 +1161,63 @@ async def async_set_dry_off(self):
         )
 
 
+class XiaomiAirHumidifierMiot(XiaomiAirHumidifier):
+    """Representation of a Xiaomi Air Humidifier (MiOT protocol)."""
+
+    @property
+    def speed(self):
+        """Return the current speed."""
+        if self._state:
+            return AirhumidifierMiotOperationMode(self._state_attrs[ATTR_MODE]).name
+
+        return None
+
+    @property
+    def button_pressed(self):
+        """Return the last button pressed."""
+        if self._state:
+            return AirhumidifierPressedButton(
+                self._state_attrs[ATTR_BUTTON_PRESSED]
+            ).name
+
+        return None
+
+    async def async_set_speed(self, speed: str) -> None:
+        """Set the speed of the fan."""
+        if self.supported_features & SUPPORT_SET_SPEED == 0:
+            return
+
+        _LOGGER.debug("Setting the operation mode to: %s", speed)
```

Comment:
This is already logged by the core.

Reply:
same
Can I fully to del the code, but which contains in models near my?

Reply:
I don't understand the question.

Reply:
Don't worry :) I'll del the fn, as told above